assets/app/scripts/controllers/services.js

```diff
@@ -24,7 +25,35 @@ angular.module('openshiftConsole')
       $scope.emptyMessage = "No services to show";
       updateFilterWarning();
       console.log("services (subscribe)", $scope.unfilteredServices);
-    }));    
+    }));
+
+    watches.push(DataService.watch("routes", $scope, function(routes){
+        $scope.routesByService = routesByService(routes.by("metadata.name"));
+        console.log("routes (subscribe)", $scope.routesByService);
+    }));
+
+    var routesByService = function(routes) {
+        var routeMap = {};
+        angular.forEach(routes, function(route){
```

Comment:
you can take a second param here which is the `routeId` instead of looking up `route.metadata.name` below